c7n/resources/ec2.py

```diff
@@ -124,6 +124,13 @@ def augment(self, resources):
         return resources
 
 
+class InstanceVolumesFilter(object):
+    """Filter instances with volumes attached"""
+
+    def filter_instances_with_volumes(self, instances):
```

Comment:
this is better as a function, in general multiple inheritance should be using sparingly, in this case its a one line function, so better as just a function, invoked from callers.